crates/theme/src/settings.rs

```diff
@@ -28,11 +28,68 @@ pub struct ThemeSettings {
     pub buffer_line_height: BufferLineHeight,
     pub active_theme: Arc<Theme>,
     pub theme_overrides: Option<ThemeStyleContent>,
+    pub themes: Option<ThemeFeatures>,
+}
+
+impl ThemeSettings {
+    pub fn reload_theme(system_is_dark: bool, cx: &mut AppContext) {
```

Comment:
Do we need to pass in `system_is_dark` if we can read it from the context?

It looks like at the call sites we always just pass in `cx.is_dark_mode()`.